feature_engine/timeseries/forecasting/base_forecast_transformers.py

```diff
@@ -78,9 +73,26 @@ def __init__(
                 f"Got {drop_original} instead."
             )
 
+        # check validity if group by variables passed
+        if group_by_variables:
+            # check group by variables data-types
+            if not (
+                isinstance(group_by_variables, str)
+                or isinstance(group_by_variables, list)
+            ):
+                raise ValueError(
+                    "group_by_variables must be an string or a list of strings. "
+                    f"Got {group_by_variables} instead."
+                )
+            # check if passed list has duplicates.
+            if isinstance(group_by_variables, list):
+                if len(set(group_by_variables)) != len(group_by_variables):
+                    raise ValueError("group_by_variables contains duplicate values")
```

Comment:
looks like tests for these 2 error catches are missing? Haven't gotten to the end of the PR so forgive me if I am wrong.

Reply:
I am going to add some test cases for them.